truffleruby/src/main/java/org/truffleruby/platform/linux/LinuxSigAction.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.platform.linux;
+
+import org.truffleruby.platform.posix.SigAction;
+
+import jnr.ffi.Runtime;
+import jnr.ffi.Struct;
+import jnr.posix.LibC.LibCSignalHandler;
+
+public final class LinuxSigAction extends SigAction {
+
+    public final Function<LibCSignalHandler> sa_handler = new Function<>(LibCSignalHandler.class);
+    public final SigSet sa_mask = inner(new SigSet());
```

Comment:
I appreciate modeling `SigSet` as a struct, but this could be a lot shorter by using:

```
public final SIgnedLong[] sa_mask = array(new SignedLong[16]);
```

If you decide to go that way instead, note that `SolarisSigAction` should be updated as well.